### DISCUSSION TITLE
How to send a differed http response

### DISCUSSION THREAD

**Discussion by luxapana** [uNetworking/uWebSockets#1302] [Q&A]
In my application, incoming http requests cannot be served synchronously. Instead, the request object is passed to a different module which evaluates the response in a worker thread. Calling thread is released immediately. The response is passed back through a different thread than the uWebSockets event loop thread. My questions:

1. when does HttpRequest and HttpResponse objects that are passed to the route callback deleted? Is it ok to keep these pointers for later use until response is sent completely?
2. Assuming that a different thread other than uSocket event loop thread cannot directly call methods in HttpResponse object, there needs to be a way to schedule a task to be run in the event loop thread. Is this supported? 

Thank you

**Comment by ghost** [comment]
1. Yes, see onAborted.
2. Loop::defer

**Reply by luxapana** [reply]
Thanks Alex.
I have few more clarifications.
Regarding, defer -> the way I understand, a loop is thread local. So if I were to call Loop::get from two threads I will get two loops. In my application, I have a dedicated thread to run the uWebSockets loop which I want to use for the defer call. The thread that tries to schedule the task through differ call is not the above thread. Could you please let me know the correct pattern to follow? 

Regarding, onAborted ->  
What is the purpose of OnAborted? 
Assuming onWritable is where I should try to send the response (for example by calling tryEnd()), how do I trigger onWritable? and how can I pass data to onWritable? Or should they be used in a different way?

**Reply by luxapana** [reply]
Hi @alexhultman Appreciate some clarification.
